packages/plugins/experimental/plugin-script/src/components/ScriptEditor.tsx

```diff
@@ -79,26 +83,30 @@ export const ScriptEditor = ({ script, role }: ScriptEditorProps) => {
         halo: client.halo,
         ownerDid,
         functionId: existingFunctionId,
-        name: script.name,
         source: script.source.content,
       });
-      if (result !== 'success' || functionId === undefined) {
+      if (result !== 'success' || functionId === undefined || functionVersionNumber === undefined) {
         throw new Error(errorMessage);
       }
       log.info('function uploaded', { functionId, functionVersionNumber });
-      const meta = getMeta(script);
+      if (fn) {
+        fn.version = functionVersionNumber;
+      }
+      const deployedFunction =
+        fn ?? space.db.add(create(FunctionType, { version: functionVersionNumber, source: script }));
+      const meta = getMeta(deployedFunction);
       setUserFunctionUrlInMetadata(meta, `/${ownerDid}/${functionId}`);
       script.changed = false;
     } catch (err: any) {
       log.catch(err);
       setError(t('upload failed label'));
     }
-  }, [existingFunctionUrl, script, script.name, script.source]);
+  }, [fn, existingFunctionUrl, script, script.name, script.source]);
 
   return (
     <div role='none' className={mx(role === 'article' && 'row-span-2', 'is-full pli-2')}>
```

Comment:
i think this far into ./components we should not be checking roles or dealing with spans. would love to start nailing this down.

Reply:
this is the top level script component